egs2/TEMPLATE/enh_asr1/setup.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+log() {
+    local fname=${BASH_SOURCE[1]##*/}
+    echo -e "$(date '+%Y-%m-%dT%H:%M:%S') (${fname}:${BASH_LINENO[0]}:${FUNCNAME[1]}) $*"
+}
+help_message=$(cat << EOF
+Usage: $0 <target-dir>
+EOF
+)
+
+
+if [ $# -ne 1 ]; then
+    log "${help_message}"
+    log "Error: 1 positional argument is required."
+    exit 2
+fi
+
+
+dir=$1
+mkdir -p "${dir}"
+
+if [ ! -d "${dir}"/../../TEMPLATE ]; then
+    log "Error: ${dir}/../../TEMPLATE should exist. You may specify wrong directory."
+    exit 1
+fi
+
+targets=""
+
+# Copy
+for f in cmd.sh conf local; do
+    target="${dir}"/../../TEMPLATE/asr1/"${f}"
+    cp -r "${target}" "${dir}"
+    targets+="${dir}/${target} "
+done
+
+
+# Symlinks to TEMPLATE
+for f in asr.sh path.sh db.sh scripts pyscripts; do
```

Comment:
For the `scripts` directory, I think you need to link to `../../TEMPLATE/enh_asr1/scripts`.
Also, the `enh_asr.sh` should be linked.

For `asr.sh`, I am not sure whether we would need it.

You could take [egs2/TEMPLATE/enh1/setup.sh#L43-L48](https://github.com/espnet/espnet/blob/master/egs2/TEMPLATE/enh1/setup.sh#L43-L48) as a reference.

Reply:
Cool. Have addressed it and the following.